### PR TITLE
feat(wren-ai-service): support sub-pipeline evaluation

### PR DESCRIPTION
**Pull request by paopa**
This PR introduces support for sub-pipeline evaluation, fixes the `extract_units` function to handle cases without a foreign key pattern, and enhances trace metadata for better identification of evaluation results.

### How to Evaluate the Sub-Pipeline

To evaluate a specific sub-pipeline, specify the pipeline name in the predict command. The evaluation command remains unchanged.

```cli
just predict <evaluation-dataset> <pipeline-name>
```

### Screenshots
image

image

### CODE REVIEW

File: wren-ai-service/eval/prediction.py [hunk 1]
```diff
@@ -237,12 +105,18 @@ def parse_args() -> Tuple[str]:
         type=str,
         help="Eval dataset file name in the eval/dataset folder",
     )
+    parser.add_argument(
```
Comment (cyyeh) [inline wren-ai-service/eval/prediction.py:108]: we can support options for users to easily understand what are supported.

https://docs.python.org/3/library/argparse.html#choices

**Review by cyyeh — COMMENTED** [review]
LGTM, only minor comment

File: wren-ai-service/eval/prediction.py (hunk 1 above)
Comment (paopa) [reply wren-ai-service/eval/prediction.py:108]: nice advice! will put in the commits.

**Review by cyyeh — APPROVED** [review]
approved

**Pull request merged**